plugins/glustershd/rest.go

```diff
@@ -124,3 +130,100 @@ func selfhealInfoHandler(w http.ResponseWriter, r *http.Request) {
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, &info.Bricks)
 
 }
+
+func selfHealHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URL
+	p := mux.Vars(r)
+	volname := p["name"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	var healType string
+	heal, ok := r.URL.Query()["type"]
+	if ok {
+		switch heal[0] {
+		case "index":
+			healType = indexHeal
+		case "full":
+			healType = fullHeal
+		default:
+			restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "heal type can only be either index or full")
+			return
+		}
+	} else {
+		healType = indexHeal
+	}
+
+	txn, err := transaction.NewTxnWithLocks(ctx, volname)
+	if err != nil {
+		if err == transaction.ErrLockTimeout {
+			restutils.SendHTTPError(ctx, w, http.StatusConflict, err)
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		}
+		return
+	}
+	defer txn.Done()
+
+	// Validate volume existence
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		if err == gderrors.ErrVolNotFound {
```

Comment:
Same here. Needs a rebase

Reply:
This comment isn't resolved.

Use ErrToStatusCode function from restutils package